### ISSUE TITLE
Create File responds with 403 when using sp-rest-proxy

### ISSUE THREAD

**Issue by theparthy** [koltyakov/sp-rest-proxy#89]
Related to Issue Created at SharePointPlus Repo
#https://github.com/Aymkdn/SharepointPlus/issues/124#issue-513256034

As requested i created a sample [repo](https://github.com/theparthy/sharepointProxyExample), where the Error occurs.

**Comment by koltyakov** [comment]
Thanks for posting the issue. I will take a look in a while.

**Comment by koltyakov** [comment]
@theparthy, I managed to make it work, SharePointPlus really uses `body.d.ListItemAllFields.__deferred.uri` to construct the next request in a chain. It's pretty opinionated and might create side effects if I add a rule for converting this in the core library.

What I'm thinking of is a way for custom ad-hoc handlers for the transformation of the payloads where it's really the case and the only case.

As a temp workaround, you can try replacing line 98 in `node_modules\sp-rest-proxy\dist\core\routers\restPost.js` with, before I come with a more robust approach:

```javascript
                .then(function (r) {
                    try {
                        if (endpointUrl.toLowerCase().indexOf('/files/add') !== -1) {
                            const body = JSON.parse(r.body);
                            if (typeof body.d.ListItemAllFields.__deferred.uri === 'string') {
                                body.d.ListItemAllFields.__deferred.uri = _this.util.buildProxyEndpointUrl(body.d.ListItemAllFields.__deferred.uri);
                            }
                            r.body = JSON.stringify(body);
                        }
                    } catch (ex) { /**/ }
                    return _this.transmitResponse(res, r);
                })
```

**Comment by theparthy** [comment]
The changes works.
Had thought of doing something similar, to circumvent the issue. 
Just did not know where to make the changes.
I'm using [patch-package](https://www.npmjs.com/package/patch-package) to have the changes modular until there is a final solution.

**Comment by koltyakov** [comment]
Hi @theparthy,

Pushed the code change which now allows injecting any response mapper for such unique cases as using SharepointPlus and rare methods that rely on absolute props values. 

Please test it out, the update is published with a beta tag `sp-rest-proxy@2.10.0-beta.2`.

## Usage sample

```typescript
const settings: IProxySettings = {
  configPath: './config/private.json',
  hooks: {
    responseMapper: (req, res, router) => {
      if (req.method === 'POST') {
        if (req.originalUrl.toLowerCase().indexOf('/files/add') !== -1) {
          const body = JSON.parse(res.body);
          if (typeof body.d.ListItemAllFields.__deferred.uri === 'string') {
            body.d.ListItemAllFields.__deferred.uri = router.util.buildProxyEndpointUrl(body.d.ListItemAllFields.__deferred.uri);
          }
          res.body = JSON.stringify(body);
        }
      }
      return res;
    }
  }
};

new RestProxy(settings).serve();
```

**Comment by koltyakov** [comment]
Going to close the issue.
